instrumentation/mongo/lib/opentelemetry/instrumentation/mongo/middlewares/subscriber.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative 'command_serializer'
+
+module OpenTelemetry
+  module Instrumentation
+    module Mongo
+      module Middlewares
+        # Event handler class for Mongo Ruby driver
+        class Subscriber
+          THREAD_KEY = :__opentelemetry_mongo_spans__
+
+          def started(event) # rubocop:disable Metrics/AbcSize
+            # start a trace and store it in the current thread; using the `operation_id`
+            # is safe since it's a unique id used to link events together. Also only one
+            # thread is involved in this execution so thread-local storage should be safe. Reference:
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring.rb#L70
+            # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring/publishable.rb#L38-L56
+            # TODO: decide naming convention
+            name = 'mongo.cmd'
+            # name = "#{event.database_name}.#{event.command_name}"
+            span = tracer.start_span(name, kind: :client)
+            set_span(event, span)
+
+            serialized_command = CommandSerializer.new(event.command).serialize
+            collection = get_collection(event.command)
+            span.set_attribute('component', 'mongo-ruby')
+            span.set_attribute('db.type', 'mongodb')
+            span.set_attribute('db.instance', event.database_name)
+            span.set_attribute('mongo.command', event.command_name)
+            span.set_attribute('mongo.collection', collection) if collection
+            span.set_attribute('db.statement', serialized_command)
+            span.set_attribute('mongo.request_id', event.request_id)
+            span.set_attribute('mongo.op_id', event.operation_id)
+            span.set_attribute('peer.hostname', event.address.host)
+            span.set_attribute('peer.port', event.address.port)
+          end
+
+          def failed(event)
+            finish_event('failed', event) do |span|
+              if event.is_a?(::Mongo::Monitoring::Event::CommandFailed)
+                span.set_attribute('error', true)
+                span.set_attribute('error.kind', 'CommandFailed')
+                span.set_attribute('message', event.message)
+              end
```

Comment:
The error should be reported in a similar fashion to https://github.com/open-telemetry/opentelemetry-ruby/blob/576bc41f45dc0286a5c4a3c01bdac5efd6a27028/sdk/lib/opentelemetry/sdk/trace/span.rb#L125-L132

In this case, I think it should be something like:
```suggestion
              if event.is_a?(::Mongo::Monitoring::Event::CommandFailed)
                add_event('exception',
                          attributes: {
                            'exception.type' => 'CommandFailed',
                            'exception.message' => event.message
                          })
              end
```